include/text.php

```diff
@@ -1271,6 +1268,22 @@ function prepare_body(&$item, $attach = false, $preview = false) {
 	$item['hashtags'] = $hashtags;
 	$item['mentions'] = $mentions;
 
+	// Compile eventual content filter reasons
+	$filter_reasons = [];
+	if (!$is_preview && !($item['self'] && local_user() == $item['uid'])) {
+		if (!empty($item['content-warning']) && (!local_user() || !PConfig::get(local_user(), 'social', 'disable_cw', false))) {
+			$filter_reasons[] = L10n::t('Content warning: %s', $item['content-warning']);
+		}
+
+		$hook_data = [
+			'item' => $item,
+			'filter_reasons' => $filter_reasons
+		];
+		Addon::callHooks('content_filter', $hook_data);
```

Comment:
In the past we named the hooks after the function in which they had been. So "prepare_body_cw" or similar would be better, I think.

Reply:
Fair enough.